src/lib/components/EventSchedule/index.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../EventScheduleModal';
+
+/**
+ * @fileoverview A schedule manager which opens and closes modals based on
+ * the current href. Looks in children for content.
+ *
+ * This does not inherit from BaseStateElement as it is not a LitElement.
+ */
+
+class EventSchedule extends HTMLElement {
+  constructor() {
+    super();
+
+    this._modalElement = document.createElement('web-event-schedule-modal');
+    this._modalElement.className = 'web-modal';
+    this._modalElement.open = false;
+    this._modalElement.addEventListener('close-modal', () => {
+      if (!window.location.hash.substr(1)) {
+        return; // we closed ourselves, so got an event that can be ignored
+      }
+      const url = window.location.pathname + window.location.search;
+      window.history.replaceState(null, null, url);
```

Comment:
It seems like pressing the back button now doesn't do anything which feels a bit confusing.
I thought it might be possible to discard the hash changes entirely so pressing the back button took you to the previous page. If that's not possible then we should probably revert to the behavior you had before where the back button reopened the modals.

Reply:
This was a bug. I think the router was stealing the behavior (I wasn't calling `stopPropagation`). It should be fixed now.

Reply:
Not a router issue—I just messed up some logic. Fixed now!